pkg/prometheus/agent/statefulset.go

```diff
@@ -137,6 +137,22 @@ func makeStatefulSet(
 				Ephemeral: ephemeral,
 			},
 		})
+	} else if storageSpec.VolumeClaimTemplates != nil {
+		for _, specVcTemplate := range storageSpec.VolumeClaimTemplates {
```

Comment:
I didn't test it but we might be able to simplify this part by removing the else block on L140 and supporting both in this one. (same comment for the server)

```suggestion
	} else {
	    if storageSpec.VolumeClaimTemplate != nil {
	        storageSpec.VolumeClaimTemplates = append(storageSpec.VolumeClaimTemplates, storageSpec.VolumeClaimTemplate)
	    }

	    for _, specVcTemplate := range storageSpec.VolumeClaimTemplates {
```

What do you think?